test/common/modules/shared_helper.js

```diff
@@ -10,299 +10,483 @@ define([
+    }
+
+    _callbackOnClose(realtime, callback) {
+      this.recordPrivateApi('read.connectionManager.activeProtocol');
+      if (!realtime.connection.connectionManager.activeProtocol) {
+        this.recordPrivateApi('call.Platform.nextTick');
+        platform.Config.nextTick(function () {
+          realtime.close();
+          callback();
+        });
+        return;
+      }
+      this.recordPrivateApi('read.connectionManager.activeProtocol.transport');
+      this.recordPrivateApi('listen.transport.disposed');
+      realtime.connection.connectionManager.activeProtocol.transport.on('disposed', function () {
+        callback();
+      });
+      /* wait a tick before closing in order to avoid the final close
+       * happening synchronously in a publish/attach callback, which
+       * complicates channelattach_publish_invalid etc. */
+      this.recordPrivateApi('call.Platform.nextTick');
       platform.Config.nextTick(function () {
         realtime.close();
-        callback();
       });
-      return;
-    }
-    realtime.connection.connectionManager.activeProtocol.transport.on('disposed', function () {
-      callback();
-    });
-    /* wait a tick before closing in order to avoid the final close
-     * happening synchronously in a publish/attach callback, which
-     * complicates channelattach_publish_invalid etc. */
-    platform.Config.nextTick(function () {
-      realtime.close();
-    });
-  }
+    }
+
+    closeAndFinishSeveral(done, realtimeArray, e) {
+      const helper = this.addingHelperFunction('closeAndFinishSeveral');
+      helper._closeAndFinishSeveral(done, realtimeArray, e);
+    }
 
-  function closeAndFinishSeveral(done, realtimeArray, e) {
-    async.map(
-      realtimeArray,
-      function (realtime, mapCb) {
-        var parallelItem = function (parallelCb) {
-          callbackOnClose(realtime, function () {
-            parallelCb();
+    _closeAndFinishSeveral(done, realtimeArray, e) {
+      async.map(
+        realtimeArray,
+        (realtime, mapCb) => {
+          var parallelItem = (parallelCb) => {
+            this.callbackOnClose(realtime, function () {
+              parallelCb();
+            });
+          };
+          mapCb(null, parallelItem);
+        },
+        function (err, parallelItems) {
+          async.parallel(parallelItems, function () {
+            if (err) {
+              done(err);
+              return;
+            }
+            done(e);
           });
+        },
+      );
+    }
+
+    /* testFn is assumed to be a function of realtimeOptions that returns a mocha test */
+    static testOnAllTransports(thisInDescribe, name, testFn, skip) {
+      const helper = this.forTestDefinition(thisInDescribe, name).addingHelperFunction('testOnAllTransports');
+      var itFn = skip ? it.skip : it;
+
+      function createTest(options) {
+        return function (done) {
+          this.test.helper = this.test.helper.withParameterisedTestTitle(name);
```

Comment:
That would mean that the `parameterisedTestHelper` field in the JSON object emitted for this usage would end up being equal to the `title`, so it wouldn't give us any extra information. The aim with `parameterisedTestTitle`

The aim of calling `withParameterisedTestTitle` is to record which _piece of code_ this test was generated by, so that we can group all of its runtime usages into a single static usage when processing the data.
in my test